features/nvstore/source/nvstore.h

```diff
@@ -0,0 +1,306 @@
+/*
+ * Copyright (c) 2018 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef __NVSTORE_H
+#define __NVSTORE_H
+
+// These addresses need to be configured according to board (in mbed_lib.json)
+#ifndef DEVICE_FLASH
+#undef NVSTORE_ENABLED
+#define NVSTORE_ENABLED 0
+#endif
+
+#if NVSTORE_ENABLED
+#include <stdint.h>
+#include <stdio.h>
+#include "platform/NonCopyable.h"
+#include "nvstore_int_flash_wrapper.h"
+#include "nvstore_shared_lock.h"
+
+enum {
+    NVSTORE_SUCCESS                =  0,
+    NVSTORE_READ_ERROR             = -1,
+    NVSTORE_WRITE_ERROR            = -2,
+    NVSTORE_NOT_FOUND              = -3,
+    NVSTORE_DATA_CORRUPT           = -4,
+    NVSTORE_BAD_VALUE              = -5,
+    NVSTORE_BUFF_TOO_SMALL         = -6,
+    NVSTORE_FLASH_AREA_TOO_SMALL   = -7,
+    NVSTORE_OS_ERROR               = -8,
+    NVSTORE_BUFF_NOT_ALIGNED       = -9,
+    NVSTORE_ALREADY_EXISTS         = -10,
+};
+
+#ifndef NVSTORE_MAX_KEYS
+#define NVSTORE_MAX_KEYS 16
+#endif
+
+#define NVSTORE_MIN_SIZE 4096
```

Comment:
Is this meant to be used? How many places is it used? If not public and not used much move to the a lesser scope please.

Reply:
Right. will move to the CPP file.